File: app/src/main/java/mx/dev/franco/automusictagfixer/utilities/shared_preferences/SharedPreferencesImpl.java
package mx.dev.franco.automusictagfixer.utilities.shared_preferences;

import android.content.Context;
import android.content.SharedPreferences;

import javax.inject.Inject;

import mx.dev.franco.automusictagfixer.utilities.Constants;


public class SharedPreferencesImpl extends AbstractSharedPreferences {
    @Inject
    SharedPreferences mSharedPreferences;

    public SharedPreferencesImpl(Context context){
        mSharedPreferences = context.getSharedPreferences(Constants.Application.FULL_QUALIFIED_NAME, Context.MODE_PRIVATE);
    }

    @Override
    public void putInt(String key, int value) {
        mSharedPreferences.edit().putInt(key,value);
        mSharedPreferences.edit().apply();
    }

    @Override
    public void putString(String key, String value) {
        mSharedPreferences.edit().putString(key,value).apply();
    }

    @Override
    public void putLong(String key, long value) {
        mSharedPreferences.edit().putLong(key,value).apply();
    }

    @Override
    public void putFloat(String key, float value) {
        mSharedPreferences.edit().putFloat(key,value).apply();
    }

    @Override
    public void putBoolean(String key, boolean value) {
        mSharedPreferences.edit().putBoolean(key,value).apply();
    }

    @Override
    public int getInt(String key) {
        return mSharedPreferences.getInt(key,-1);
    }

    @Override
    public String getString(String key) {
        return mSharedPreferences.getString(key, null);
    }

    @Override
    public long getLong(String key) {
        return mSharedPreferences.getLong(key,-1);
    }

    @Override
    public float getFloat(String key) {
        return mSharedPreferences.getFloat(key,-1);
    }

    @Override
    public boolean getBoolean(String key) {
        return mSharedPreferences.getBoolean(key,false);
    }

    @Override
    public void remove(String key) {
        mSharedPreferences.edit().remove(key).apply();
    }
}

File: app/src/main/java/mx/dev/franco/automusictagfixer/utilities/shared_preferences/SharedPreferencesImpl.kt
package mx.dev.franco.automusictagfixer.utilities.shared_preferences

import android.content.Context
import android.content.SharedPreferences
import mx.dev.franco.automusictagfixer.utilities.Constants

class SharedPreferencesImpl() : AbstractSharedPreferences() {
    private var mSharedPreferences: SharedPreferences? = null

    constructor(context: Context) : this() {
        mSharedPreferences = context.getSharedPreferences(
            Constants.Application.FULL_QUALIFIED_NAME,
            Context.MODE_PRIVATE
        )
    }

    override fun putInt(key: String?, value: Int) {
        mSharedPreferences!!.edit().putInt(key, value).apply()
    }

    override fun putString(key: String?, value: String?) {
        mSharedPreferences!!.edit().putString(key, value).apply()
    }

    override fun putLong(key: String?, value: Long) {
        mSharedPreferences!!.edit().putLong(key, value).apply()
    }

    override fun putFloat(key: String?, value: Float) {
        mSharedPreferences!!.edit().putFloat(key, value).apply()
    }

    override fun putBoolean(key: String?, value: Boolean) {
        mSharedPreferences!!.edit().putBoolean(key, value).apply()
    }

    override fun getInt(key: String?): Int {
        return mSharedPreferences!!.getInt(key, -1)
    }

    override fun getString(key: String?): String? {
        return mSharedPreferences!!.getString(key, null)
    }

    override fun getLong(key: String?): Long {
        return mSharedPreferences!!.getLong(key, -1)
    }

    override fun getFloat(key: String?): Float {
        return mSharedPreferences!!.getFloat(key, -1f)
    }

    override fun getBoolean(key: String?): Boolean {
        return mSharedPreferences!!.getBoolean(key, false)
    }

    override fun remove(key: String?) {
        mSharedPreferences!!.edit().remove(key).apply()
    }
}